Makefile

```diff
@@ -111,6 +112,18 @@ ifeq "$(LXD_OFFLINE)" ""
 endif
 	swagger generate spec -o doc/rest-api.yaml -w ./lxd -m
 
+.PHONY: doc
+doc:
+	@echo "Setting up documentation build environment"
+	python3 -m venv .sphinx/venv
+	. $(SPHINXENV) ; pip install --upgrade -r .sphinx/requirements.txt
+	mkdir -p .sphinx/deps/ .sphinx/themes/
+	git -C .sphinx/deps/vanilla pull || git clone https://github.com/evildmp/vanilla-sphinx-test .sphinx/deps/vanilla
+	ln -sf ../deps/vanilla/vanilla .sphinx/themes/vanilla
+	@echo "Build the documentation"
+	rm -Rf doc/html/
+	. $(SPHINXENV) ; sphinx-build -c .sphinx/ -b dirhtml doc/ doc/html/ -w .sphinx/warnings.txt
```

Comment:
This should be okay for now, but I'm a bit concerned about the ``-c .sphinx/``. It might prevent us from using convenient paths like "/image/figure.png" or "/reuse/file.md" in the future, forcing us to use relative paths (which are error prone).